### PR TITLE
upgrade(contract): Update contracts to 0.6

### PR DESCRIPTION
**Pull request by mfornet**
Update solidity contracts version and compile with linked version to rainbow-bridge-sol in Github

### CODE REVIEW

File: erc20-connector/contracts/BridgeToken.sol [hunk 1]
```diff
@@ -1,10 +1,10 @@
-pragma solidity ^0.5.0;
+pragma solidity ^0.6.0;
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 
 contract BridgeToken is ERC20 {
     address private _controller; // has CONTROL role
 
-    constructor() public {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) public {
```
Comment (mfornet) [inline erc20-connector/contracts/BridgeToken.sol:7]: In version it is required calling constructor of inherited contract.

File: erc20-connector/contracts/BridgeTokenFactory.sol [hunk 2]
```diff
@@ -56,7 +56,7 @@ contract BridgeTokenFactory is ERC20Locker {
 
     function newBridgeToken(string calldata nearTokenId) external returns (BridgeToken) {
         require(!_isBridgeToken[_nearToEthToken[nearTokenId]], "ERR_BRIDGE_TOKEN_EXISTS");
-        BridgeToken btoken = new BridgeToken();
+        BridgeToken btoken = new BridgeToken("Rainbow", "RB");
```
Comment (mfornet) [inline erc20-connector/contracts/BridgeTokenFactory.sol:59]: Do we want to have this default values or do we prefer using something like: `f"rainbow-{nearTokenId}"`

File: erc20-connector/package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
         "eslint-plugin-standard": "^4.0.0",
         "eth-gas-reporter": "^0.2.17",
         "ganache-cli": "^6.9.1",
-        "solc": "^0.5.17",
+        "solc": "0.6.2",
```
Comment (mfornet) [inline erc20-connector/package.json:25]: This particular version was chosen for several reasons:

- There are some dependencies that use `0.6.2` (@openzeppelin/contracts/utils/Address.sol) so we can't use any lower version
- For larger version (>= 0.6.8) there is an issue with flat files and multiple licenses. It throw an error (read more here: https://forum.openzeppelin.com/t/solidity-0-6-8-introduces-spdx-license-identifiers/2859) (We need to file an alternative to the current tool we are using to flatten files)

**Comment by mfornet** [comment]
See comments in the code about some changes and their motivation. Everything was precompiled calling `yarn` first and then `./dist.sh` on `erc20-connector` directory.

Notice that `erc20-connector/dist/Migrations.full.abi` and ` erc20-connector/dist/Migrations.full.bin` was not generated (there were not error so I'm not sure why) but the equivalente files in `res` folder were created.

**Comment by nearmax** [comment]
I am getting the following warnings when compiling:
```
:~/rainbow-token-connector/erc20-connector$ ./dist.sh 
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/BridgeTokenFactory.full.sol -o dist
dist/BridgeTokenFactory.full.sol:774:5: Warning: Function state mutability can be restricted to pure
    function decodeInitialValidators(Borsh.Data memory data) internal view returns(InitialValidators memory validators) {
    ^ (Relevant source part starts here and spans across multiple lines).

Done in 15.32s.
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/BridgeToken.full.sol -o dist
Done in 4.19s.
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/ERC20Locker.full.sol -o dist
dist/ERC20Locker.full.sol:774:5: Warning: Function state mutability can be restricted to pure
    function decodeInitialValidators(Borsh.Data memory data) internal view returns(InitialValidators memory validators) {
    ^ (Relevant source part starts here and spans across multiple lines).

Done in 14.81s.
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/Locker.full.sol -o dist
dist/Locker.full.sol:473:5: Warning: Function state mutability can be restricted to pure
    function decodeInitialValidators(Borsh.Data memory data) internal view returns(InitialValidators memory validators) {
    ^ (Relevant source part starts here and spans across multiple lines).

Done in 9.09s.
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/Migrations.full.sol -o dist
Done in 3.84s.
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/NearProverMock.full.sol -o dist
dist/NearProverMock.full.sol:15:27: Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    function proveOutcome(bytes memory proofData, uint64 blockHeight) public override view returns(bool) {
                          ^--------------------^

dist/NearProverMock.full.sol:15:51: Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    function proveOutcome(bytes memory proofData, uint64 blockHeight) public override view returns(bool) {
                                                  ^----------------^

Done in 14.16s.
yarn run v1.22.5
$ /home/max/rainbow-token-connector/erc20-connector/node_modules/.bin/solcjs --bin --abi --optimize dist/TToken.full.sol -o dist
Done in 12.32s.
```

File: erc20-connector/package.json (hunk 3 above)
Comment (abacabadabacaba) [reply erc20-connector/package.json:25]: You can add the same hack that we are using [here](https://github.com/near/rainbow-bridge/blob/master/contracts/eth/dist.sh#L19).

**Comment by mfornet** [comment]
Done in #20

**Pull request closed without merging**